utils/variables.go

```diff
@@ -9,13 +9,17 @@ import (
 )
 
 // ValidNameRegex is the pattern that matches to a valid name.
-// The name must begin with a letter i.e. [a-zA-Z],
-// and the body can only contain 0 or more numbers, letters, dashes and underscores i.e. [-\w]*.
-var ValidNameRegex = regexp.MustCompile(`^[a-zA-Z][-\w]*$`)
+// The name must begin with a letter or number i.e. [a-zA-Z0-9],
+// and can only contain up to 60, letters, numbers, dashes, and underscores i.e. [-\w]*.
+var ValidNameRegex = regexp.MustCompile(`^[a-zA-Z0-9]([-\w]){0,59}$`)
```

Comment:
I know the scope was approved, but I did want to call attention to the fact that this constraint of 60 characters could* break existing configs. The input which surfaced this catch (nice one Zack!) is 45 characters:
```
$ echo -n "0b43296e-85fe-49eb-805a-f5026f169266_receipts" | wc -c
45
```

The idea that some resource names are partially machine generated (I think? I'm not positive on how org UUIDs and viam packages and name validation are all coming together) was new to me.

If a human typed 60 characters into some input field and suddenly hit this clause, I expect them to be surprised but understand how to chase this down and address it. But this showing up the way it did makes me worried enough to ask about backwards compatibility.

Ultimately, I don't have enough context to assess the risk of annoying a customer. I assume if we were hyper conservative we could scan the robot configs in the database?

So if everyone is bought into the risk being acceptable I'm good with this!